dispatch.go

```diff
@@ -125,6 +124,7 @@ func (d *dispatch) handleCall(calls map[int]*call, c *call) {
 	go func() {
 		select {
 		case <-c.ctx.Done():
+			c.ch <- newCanceledError(c.method, seqid)
```

Comment:
All of this is tested in `protocol_test.go:TestLongCallCancel`. I validate that:

1. The client has canceled the call and responded with an empty response (and `CanceledError`).
2. The server has received the cancellation and successfully cancelled the call.
3. When we don't cancel a call, the correct response is returned.

I don't explicitly assert that a reply for a canceled wasn't sent, although a coverage analysis shows that that code was exercised (and because there's a race there). I also don't explicitly validate the dropped cancel dispatch (after client cancel), because I'm not sure what we would hope to gain from that test. Its results follow from the other tests.